### ISSUE TITLE
`from csv` returns a malformed error 

### ISSUE THREAD

**Issue by KAAtheWiseGit** [nushell/nushell#11722]
### Describe the bug

Trying to open a moderately large (10K lines, 2.4MB) semicolon-separated CSV file returns weird non-error output:

```Nushell
~/loads/firefox> open HW1_var_1.csv
Couldn't fit table into 174 columns!Couldn't fit table into 174 columns!Couldn't fit table into 174 columns!Couldn't fit table into 174 columns!Couldn't fit table into 174 columns!Couldn't fit table into 174 columns!Couldn't fit table into 174 columns!Couldn't fit table into 174 columns!Couldn't fit table into 174 columns!Couldn't fit table into 174 columns!Couldn't fit table into 174 columns!
```

### How to reproduce

-

### Expected behavior

A proper `ShellError`.

### Configuration

| key                | value                                                        |
| ------------------ | ------------------------------------------------------------ |
| version            | 0.89.0                                                       |
| branch             |                                                              |
| commit_hash        |                                                              |
| build_os           | linux-x86_64                                                 |
| build_target       | x86_64-alpine-linux-musl                                     |
| rust_version       | rustc 1.72.1 (d5c2e9c34 2023-09-13) (Alpine Linux 1.72.1-r0) |
| cargo_version      | cargo 1.72.1                                                 |
| build_time         | 2024-01-13 03:02:49 +00:00                                   |
| build_rust_channel | release                                                      |
| allocator          | mimalloc                                                     |
| features           | default, sqlite, trash, which, zip                           |
| installed_plugins  |                                                              |

**Comment by fdncred** [comment]
Right, csv is `comma` separated values, not semicolon separated values. The error could be improved, I'm sure.

**Comment by KAAtheWiseGit** [comment]
Ah, it's not the CSV command.  It dutifully turns the entire thing into a table.  It's the table viewer

```Rust
fn report_unsuccessful_output(ctrlc1: Option<Arc<AtomicBool>>, term_width: usize) -> String {
    if nu_utils::ctrl_c::was_pressed(&ctrlc1) {
        "".into()
    } else {
        // assume this failed because the table was too wide
        // TODO: more robust error classification
        format!("Couldn't fit table into {term_width} columns!")
    }
}
```

**Comment by fdncred** [comment]
Right, I totally missed that. I didn't look close enough at the error. CSV is trying to put your whole file in one column, LOL.
